pandas/core/algorithms.py

```diff
@@ -221,8 +221,8 @@ def sort_mixed(values):
         ordered = sort_mixed(values)
     else:
         try:
-            sorter = values.argsort()
-            ordered = values.take(sorter)
+            sorter = _ensure_int64(values.argsort())
+            ordered = take_nd(values, sorter, allow_fill=False)
```

Comment:
I would do as many ensures inside take_nd that u can - so we can just call it with anything and it will work